imports/server/methods/sendChatMessageV2.ts

```diff
@@ -0,0 +1,45 @@
+import { check, Match } from 'meteor/check';
+import { Meteor } from 'meteor/meteor';
+import { ChatMessageContentType } from '../../lib/schemas/ChatMessage';
+import sendChatMessageV2 from '../../methods/sendChatMessageV2';
+import sendChatMessageInternalV2 from '../sendChatMessageInternalV2';
+
+sendChatMessageV2.define({
+  validate(arg) {
+    check(arg, {
+      puzzleId: String,
+      content: String,
+    });
+
+    return arg;
+  },
+
+  async run({ puzzleId, content }: { puzzleId: string, content: string }) {
+    check(this.userId, String);
+    const contentObj = JSON.parse(content);
+    check(contentObj, {
+      type: String,
+      children: [Match.OneOf({
+        type: String,
```

Comment:
Can we just check for a literal `'mention'` here instead of scanning below?

Reply:
The docs for `check` do not suggest they support value literals.  I guess I could use a `Match.Where(condition)` instead?

Reply:
Huh. I...am pretty sure I didn't make that up, but I agree it's not documented behavior. We've certainly relied on it in other places we've used `Match.OneOf`, though, and the code seems to explicitly intend to support it: https://github.com/meteor/meteor/blob/9cc480a1ab3bfc85ebc86a86a85bfa92a19103b9/packages/check/match.js#L240-L250

```
> const pat = { type: 'mention' };
> check({ type: 'mention' }, pat)
> check({ type: 'not-a-mention' }, pat);
Uncaught:
errorClass [Error]: Match error: Expected mention, got "not-a-mention" in field type
```

Reply:
Oh, the docs don't say it, but it works, and if I throw in an `as const` it even gets correctly inferred as being compatible with `ChatMessageContentType`.  Fixed in afc4e4e78da6f8c07dd9e0716cbeecbe68d1f72e